### DISCUSSION TITLE
VIEW Invalid query - invalid identifier parts

### DISCUSSION THREAD

**Discussion by strash** [pocketbase/pocketbase#4778] [Q&A] [answered]
Hi, it's me again with weird queries.

What I'm trying to achieve is to return `[]` instead `['null']` inside the projects column by using case statement.

```sql
SELECT
  r.id,
  COUNT(DISTINCT p.id) AS count,
  CASE
    WHEN COUNT(DISTINCT p.id) = 0
      THEN JSON_ARRAY()
    ELSE JSON_GROUP_ARRAY(DISTINCT p.id)
  END AS projects
FROM project_members AS pm, JSON_EACH(pm.roles) AS jpmr
JOIN projects AS p on pm.project = p.id AND p.is_public IS TRUE
RIGHT JOIN roles AS r ON jpmr.VALUE = r.id
GROUP BY r.id;
```

<img width="679" alt="image" src="https://github.com/pocketbase/pocketbase/assets/14233263/3edbcc73-189e-42d3-8569-70a525c2b612">

### But I'm getting this error.

<img width="930" alt="image" src="https://github.com/pocketbase/pocketbase/assets/14233263/ed453402-0564-488f-9fc4-bb3d98badf08">

### How it should be:

<img width="609" alt="image" src="https://github.com/pocketbase/pocketbase/assets/14233263/bd4e72eb-6dc8-4855-8654-83f080733e91">

### I double checked how I run the server and CGO is not enabled.
`CGO_ENABLED=0 go run main.go serve --dev=true --http=localhost:3000`

### Working but not ideal query without case statement

```sql
SELECT
  r.id,
  COUNT(DISTINCT p.id) AS count,
  JSON_GROUP_ARRAY(DISTINCT p.id) AS projects
FROM project_members AS pm, JSON_EACH(pm.roles) AS jpmr
JOIN projects AS p on pm.project = p.id AND p.is_public IS TRUE
RIGHT JOIN roles AS r ON jpmr.VALUE = r.id
GROUP BY r.id;
```

<img width="605" alt="image" src="https://github.com/pocketbase/pocketbase/assets/14233263/e55fb5f8-d66d-4a89-81aa-7997d0920665">

**Comment by ganigeorgiev** [comment, marked as answer]
This will be documented better in the future. Try to wrap the expression in parenthesis, aka.:

```sql
(CASE ... END) as projects
```

**Reply by strash** [reply]
It works! Thank you for your quick response!
